packages/paste-core/utilities/flex/src/index.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Box} from '@twilio-paste/box';
+import {FlexboxProps} from '@twilio-paste/types';
+
+export type displayOptions = 'flex' | 'inline-flex';
+export type vAlignOptions = 'top' | 'center' | 'bottom';
+export type hAlignOptions = 'left' | 'center' | 'right';
+
+interface Flex {
+  display?: displayOptions;
+  column?: boolean;
+  vAlignContent?: vAlignOptions;
+  hAlignContent?: hAlignOptions;
+  grow?: boolean | number;
+  shrink?: boolean | number;
+  basis?: string | number;
+  order?: number;
+  wrap?: boolean;
+}
+
+function getGrow(props: Flex): number {
+  const {grow} = props;
+  if (typeof grow === 'number') {
+    return grow;
+  } else if (grow) {
+    return 1;
+  }
+
+  return 0; // default
+}
+
+function getShrink(props: Flex): number {
+  const {shrink, basis} = props;
+  if (typeof shrink === 'number') {
+    return shrink;
+  } else if (shrink) {
+    return 1;
+  } else if (shrink === false) {
+    return 0;
+  }
+
+  if (basis && basis !== 'auto') {
+    return 0;
+  }
+
+  return 1; // default
+}
+
+function getBasis(props: Flex): string {
+  const {basis} = props;
+  if (basis) {
+    const suffix = typeof basis === 'number' || String(parseInt(basis as string, 10)) === basis ? 'px' : '';
+    return basis + suffix;
+  }
+
+  return 'auto'; // default
+}
+
+function getFlexStyles(props: Flex): FlexboxProps {
+  const {column, wrap, vAlignContent, hAlignContent} = props;
+
+  function alignPropToFlex(align: Flex['vAlignContent'] | Flex['hAlignContent']) {
+    switch (align) {
+      case 'top':
+      case 'left':
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'bottom':
+      case 'right':
+        return 'flex-end';
+    }
+  }
+
+  return {
+    flex: `${getGrow(props)} ${getShrink(props)} ${getBasis(props)}`,
+    flexDirection: column ? 'column' : 'row',
+    flexWrap: wrap ? 'wrap' : 'nowrap',
+    justifyContent: alignPropToFlex(column ? vAlignContent : hAlignContent),
+    alignItems: alignPropToFlex(column ? hAlignContent : vAlignContent),
+  };
+}
+
+const Flex: React.FC<Flex> = props => {
+  return (
+    <Box {...getFlexStyles(props)} display={props.display} order={props.order}>
```

Comment:
I'm worried about supporting order as when it is used to change the visual appearance from then natural flow of the DOM, it will cause accessibility failures. Visually elements should appear in the same order as you would progress through the DOM with a screen reader or keyboard focus, in the direction that is suited to language.

So in English pages should visually traverse left to right, top to bottom as you tab through the page. Covered under [logical focus order](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)

Reply:
Good point.  I've never really used it myself in live code and always thought of it as a crutch for bad markup or SEO hackery which doesn't matter in 2019.  I'm +1 for removing it.

Reply:
Agreed on this. Removing.